Source/Core/GregorianDate.js

```diff
@@ -69,22 +70,57 @@ function GregorianDate(
   this.isLeapSecond = isLeapSecond;
 
   function validateRange() {
-    if (year < 1 || year > 9999)
+    const maxYear = 9999;
+    const minYear = 1;
+    const minMonth = 1;
+    const maxMonth = 12;
+    const maxDay = 31;
+    const minDay = 1;
+    const minHour = 0;
+    const maxHour = 23;
+    const maxMinute = 59;
+    const minMinute = 0;
+    const minSecond = 0;
+    const maxSecond = 59;
+    const minMilisecond = 0;
+    const excludedMaxMilisecond = 1000;
+
+    if (year < minYear || year > maxYear)
       throw "Year parameter represent an invalid date";
-    if (month < 1 || month > 12)
+    if (month < minMonth || month > maxMonth)
       throw "Month parameter represent an invalid date";
-    if (day < 1 || day > 31) throw "Day parameter represent an invalid date";
-    if (hour < 0 || hour > 23) throw "Hour parameter represent an invalid date";
-    if (minute < 0 || minute > 59)
+    if (day < minDay || day > maxDay)
+      throw "Day parameter represent an invalid date";
+    if (hour < minHour || hour > maxHour)
+      throw "Hour parameter represent an invalid date";
+    if (minute < minMinute || minute > maxMinute)
       throw "Combination of Minute and IsLeapSecond parameters represent an invalid date";
     if (
-      second < 0 ||
-      (second > 59 && !isLeapSecond) ||
-      (second > 60 && isLeapSecond)
+      second < minSecond ||
+      (second > maxSecond && !isLeapSecond) ||
+      (second > maxSecond + 1 && isLeapSecond)
     )
       throw "Second parameter represent an invalid date";
-    if (millisecond < 0 || millisecond >= 1000)
+    if (millisecond < minMilisecond || millisecond >= excludedMaxMilisecond)
       throw "Millisecond parameter represent an invalid date";
   }
+
+  // Javascript date object supports only dates greater than 1901. Thus validating with custom logic
+  function validateDate() {
+    const minNumberOfDaysInMonth = 28;
+    const monthsWith31Days = [1, 3, 5, 7, 8, 10, 12];
```

Comment:
This may be nitpicky, but can we handle the amount of days in each month [the same way that `JulianDate.js` handles them](https://github.com/CesiumGS/cesium/blob/6bbf84b7d0bc734ff3c7678830fb19899675950d/Source/Core/JulianDate.js#L12)? You would still have to check for the Feburary leap year case, but I think this should simplify the rest of the code.

Reply:
Refactored this method to make it simpler and more readable.